src/shared/varlink.c

```diff
@@ -579,11 +579,11 @@ static int varlink_parse_message(Varlink *v) {
 
         sz = e - begin + 1;
 
-        varlink_log(v, "New incoming message: %s", begin);
+        varlink_log(v, "New incoming message: %s", begin); /* FIXME: should we output the whole message here before validation? */
 
         r = json_parse(begin, 0, &v->current, NULL, NULL);
         if (r < 0)
-                return r;
+                varlink_log_errno(v, r, "Failed to parse JSON: %m");
```

Comment:
hmmm, maybe do:

```c
if (r < 0) {
        /* If we encounter a parse failure flush all data. We cannot possibly recover from this, hence drop all buffered data now */
        v->input_buffer_index = v->input_buffer_size = v->input_buffer_unscanned = 0;
        return varlink_log_errno(v, r, …);
}
```

i.e. we cannot recover from parse errors anyway, and I think tere's no need to be precise and just drop what we couldn't handle, but drop everything.